### ISSUE TITLE
Ranking de la tarea

### ISSUE THREAD

**Issue by estephany-up** [estephany-up/Incentivos_parte_2#1]
Hola! Tengo una duda con respecto al código que puedo usar para el ranking de la tarea. El código que pensaba plantear es el siguiente:
def rank(self):
       ##i_1 representa al puesto uno
        i_1=0
        i_2=0
        i_3=0
        i_4=0

        for p in self.group.get_players(): 
            if p.total_answer_correct_R1 > i_1:
                i_1=p.total_answer_correct_R1
            else:
                i_2=p.total_answer_correct_R1

Sin embargo, creo que lo que trato de plantear sólo funciona para dos players y mi intención es que funcione para 4 (esta es la cantidad de jugadores en un grupo). Por ello, quería saber si es que se puede guardar el id de cada uno de los participantes del grupo en una variable como por ejemplo p_a y luego usar lo siguiente p_a.total_answer_correct_R1. Y usar esto último para comparar con un if y no usar el for, de ahí recién guardar los puntajes de cada player, pero en variables que sean reconocibles y usarlas directamente en el html. Preguntaba si es posible hacer esto: p_a.total_answer_correct_R1 porque evitaría definir muchas variables. Dado que el otro caso sería guardar el puntaje de cada jugador del grupo en una variable y después usar eso para hacer la comparación con el if y guardarlo en otras variables en el orden en el cual irían en el ranking.

**Comment by Sergio2405** [comment]
Sii se puede obtener el id, lo podrias obtener de la siguiente forma (entiendo que te refieres a esto): 
```
p1 = self.get_player_by_id(1)
p2 = self.get_player_by_id(2)
Lo que esta dentro del parentesis se refiere al jugador i , i = {1,2,3,4}
```
@estephany-up

**Comment by estephany-up** [comment]
Claro, pero en mi caso lo que quiero es hacer un ranking de los puntajes (de mayor a menor) y al lado poner el id de cada uno de los participantes, todo esto en una tabla. Entonces, replantee lo de arriba de la siguiente manera:
## esto va en class Ranking_R1(Page):
def vars_for_template(self):
        i_1=0 ##puestos 
        i_2=0
        i_3=0
        i_4=0
        p_1=0 #id del puesto
        p_2=0
        p_3=0
        p_4=0

        for p in self.group.get_players(): 
            if p.total_answers_correct_R1 > i_4:  ##total_answer_correct_R1 se crea en player y en page Tarea_conteo_R1 se llena con un valor
                i_4=p.total_answers_correct_R1
                p_4=p.id_in_group()
            if i_4 > i_3:
                temp=i_3
                i_3=i_4
                i_4=temp
                tee=p_3
                p_3=p_4
                p_4=tee
            if i_3 > i_2:
                temp=i_2
                i_2=i_3
                i_3=temp
                tee=p_2
                p_2=p_3
                p_3=tee
            if i_2 > i_1:
                temp=i_1
                i_1=i_2
                i_2=temp
                tee=p_1
                p_1=p_2
                p_2=tee
        return dict(
            pt_1=i_1, pt_2=i_2, pt_3=i_3, pt_4=i_4,
            p_1=p_1, p_2=p_2, p_3=p_3, p_4=p_4,
        )
Y en el template Tarea_conteo_R1 hago mi tabla con los pt_s y p_s.
A la hora de correr no funciona, me sale que esto: p.total_answers_correct_R1 es Non-type por lo tanto no puedo comparar con i_4 que es un interger. Aún me falta modificar la función e indicar lo que pasaría cuando ambos puntajes son iguales, pero esto es lo esencial.

**Comment by Sergio2405** [comment]
Pueden probar corriendo ese for en models. Y que les retorne una lista con los pt_s y los p_s 
@estephany-up

**Comment by estephany-up** [comment]
Intenté correrlo la función estando en models y si corrió. Pero, en la página del template, la tabla aparecía con varios None. Bueno, creo que era porque me faltaba poner return. Después de ello, lo hice con listas como me recomendaste y tampoco tuve suerte.
Esto va en players:
def rank_R1(self):
        i_1=0 ##puestos 
        i_2=0
        i_3=0
        i_4=0
        r_1=0 ##id player
        r_2=0
        r_3=0
        r_4=0

        for p in self.group.get_players(): 
            if p.total_answers_correct_R1 > i_4:
                i_4=p.total_answers_correct_R1
                r_4=p
            if i_4 > i_3:
                temp=i_3
                i_3=i_4
                i_4=temp
                tee=r_3
                r_3=r_4
                r_4=tee
            if i_3 > i_2:
                temp=i_2
                i_2=i_3
                i_3=temp
                tee=r_2
                r_2=r_3
                r_3=tee
            if i_2 > i_1:
                temp=i_1
                i_1=i_2
                i_2=temp
                tee=r_1
                r_1=r_2
                r_2=tee

        return [i_1, i_2, i_3, i_4], [r_1, r_2, r_3, r_4]
Y esto es lo que estado intentando poner en pages:
def vars_for_template(self):
        pt_p, p_p = self.player.rank_R1()
        p_p1=p_p[1]
        p_p2=p_p[2]
        p_p3=p_p[3]
        p_p4=p_p[4]
        pt_p1=pt_p[1]
        pt_p2=pt_p[2]
        pt_p3=pt_p[3]
        pt_p4=pt_p[4]
        return dict(
            p_p1=p_p1, pt_p1=pt_p1,
            p_p2=p_p2, pt_p2=pt_p2,
            p_p3=p_p3, pt_p3=pt_p3,
            p_p4=p_p4, pt_p4=pt_p4
        )
He reformulado todo de todas las formas que se me ocurrían en el momento, pero no tuve suerte.

**Comment by Sweetdreamstome** [comment]
Hola chicas!! Otra forma de hacerlo, en vez de ponerlo en players, pónganlo en groups, ya que esta función engloba a varios players de por si, no solamente a un player. Cambien sus variables como vean conveniente (si es que necesitan un cambio) y para el vars_for_template de pages, tendrían que poner: self.group.rank_R1().
